### ISSUE TITLE
Row hash validation of custom SQL query not working correctly

### ISSUE THREAD

**Issue by danieldeleo** [GoogleCloudPlatform/professional-services-data-validator#806]
For the given source Teradata SQL:

```
SEL
  s_name,
  s_acctbal,
  RANK() OVER (ORDER BY s_acctbal DESC) AS acct_rank
FROM tpch.supplier
QUALIFY RANK() OVER (ORDER BY s_acctbal DESC) <= 10;
```
and target BigQuery SQL:

```
SELECT
    supplier.s_name,
    supplier.s_acctbal,
    rank() OVER (ORDER BY supplier.s_acctbal DESC) AS acct_rank
  FROM
    `ddeleo-bigshift-demo`.edw_data.supplier
  QUALIFY acct_rank <= 10;
```

Running the following DVT command:

```
data-validation validate custom-query
  --source-conn Teradata_CONN 
  --target-conn BigQuery_CONN 
  --custom-query-type row 
  --source-query-file source_teradata/select_qualify.sql
  --target-query-file target_teradata/select_qualify.sql 
  --primary-keys s_name \
  --bq-result-handler ddeleo-bigshift-demo.bigshift_logs.bigshift_dvt_results 
  --hash '*' 
  -l unique_id=e1698985-1e4f-4b0b-9be0-bd20435131c5,file_name=teradata/select_qualify.sql
```

Results in the following error:

```
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO - Traceback (most recent call last):
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/bin/data-validation", line 8, in <module>
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     sys.exit(main())
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/__main__.py", line 566, in main
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     validate(args)
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/__main__.py", line 542, in validate
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     run(args)
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/__main__.py", line 502, in run
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     run_validations(args, config_managers)
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/__main__.py", line 481, in run_validations
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     run_validation(config_manager, verbose=args.verbose)
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/__main__.py", line 470, in run_validation
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     validator.execute()
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/data_validation.py", line 97, in execute
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     result_df = self._execute_validation(
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/data_validation.py", line 292, in _execute_validation
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     target_query = validation_builder.get_target_query()
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/validation_builder.py", line 385, in get_target_query
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     query = self.target_builder.compile(self.validation_type, table)
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/query_builder/query_builder.py", line 493, in compile
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     table = table.mutate(self.compile_calculated_fields(table, n))
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/query_builder/query_builder.py", line 465, in compile_calculated_fields
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     return [
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/query_builder/query_builder.py", line 466, in <listcomp>
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     field.compile(table)
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/query_builder/query_builder.py", line 399, in compile
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     compiled_fields = self._compile_fields(ibis_table, self.fields)
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/data_validation/query_builder/query_builder.py", line 395, in _compile_fields
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     compiled_fields.append(ibis_table[field])
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/ibis/expr/types.py", line 439, in __getitem__
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     return self.get_column(what)
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/ibis/expr/types.py", line 553, in get_column
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     ref = ops.TableColumn(name, self)
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/ibis/expr/operations.py", line 224, in __init__
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     super().__init__(name, table)
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/ibis/expr/signature.py", line 183, in __init__
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     self._validate()
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -   File "/usr/local/lib/python3.9/site-packages/ibis/expr/operations.py", line 228, in _validate
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO -     raise com.IbisTypeError(
[2023-04-13, 21:39:32 UTC] {pod_manager.py:237} INFO - ibis.common.exceptions.IbisTypeError: 'S_NAME' is not a field in ['s_name', 's_acctbal', 'acct_rank']
```

The error:
`ibis.common.exceptions.IbisTypeError: 'S_NAME' is not a field in ['s_name', 's_acctbal', 'acct_rank']`

is a case sensitivity issue [occurring here](https://github.com/ibis-project/ibis/blob/98e2cd84478f9c8325879bb5b77527a23145f419/ibis/expr/operations.py#L226) and could be rooted in using an older Ibis version 1.4.0 which DVT [currently uses](https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/ad0e64f4356394826fabd73d1a279ed448cb8273/setup.py#L35) 

This may be a non-issue in later versions of Ibis

**Comment by danieldeleo** [comment]
This same behavior occurs for simple select queries as well (no QUALIFY clause, just `select col from table`)

**Comment by nehanene15** [comment]
I believe this should be resolved now with the Ibis upgrade PR #894 - if this is still an issue we can re-open.
